apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/Content/useRefetchStats.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback } from 'react'
+
+import {
+  DocumentVersion,
+  Evaluation,
+  EvaluationResultableType,
+} from '@latitude-data/core/browser'
+import { useCurrentCommit } from '@latitude-data/web-ui'
+import useEvaluationResultsCounters from '$/stores/evaluationResultCharts/evaluationResultsCounters'
+import useEvaluationResultsMeanValue from '$/stores/evaluationResultCharts/evaluationResultsMeanValue'
+import useEvaluationResultsModalValue from '$/stores/evaluationResultCharts/evaluationResultsModalValue'
+import useAverageResultsAndCostOverCommit from '$/stores/evaluationResultCharts/numericalResults/averageResultAndCostOverCommitStore'
+import useAverageResultOverTime from '$/stores/evaluationResultCharts/numericalResults/averageResultOverTimeStore'
+
+export function useRefetchStats({
+  evaluation,
+  document,
+}: {
+  evaluation: Evaluation
+  document: DocumentVersion
+}) {
+  const { commit } = useCurrentCommit()
+  const evaluationId = evaluation.id
+  const commitUuid = commit.uuid
+  const documentUuid = document.documentUuid
+  const isNumeric =
+    evaluation.configuration.type === EvaluationResultableType.Number
+
+  const { refetch: refetchAverageResulstAndCostsOverCommit } =
+    useAverageResultsAndCostOverCommit({
+      evaluation,
+      documentUuid,
+    })
+  const { refetch: refetchAverageResultOverTime } = useAverageResultOverTime({
+    evaluation,
+    documentUuid,
+  })
+  const { refetch: refetchMean } = useEvaluationResultsMeanValue({
+    commitUuid,
+    documentUuid,
+    evaluationId,
+  })
+  const { refetch: refetchModal } = useEvaluationResultsModalValue({
+    commitUuid,
+    documentUuid,
+    evaluationId,
+  })
+  const { refetch: refetchTotals } = useEvaluationResultsCounters({
```

Comment:
is this refetch actually the mutate function? if so i wouldn't rename it it's better to expose the real name so devs understand what they can do with this

Reply:
If this function just triggers a new request to update the data, I agree with @andresgutgon here ☝️

Reply:
but it doesn't just do that, you can also pass it data to mutate the store

Reply:
In general yes, but in this use case is a refetch. We're talking semantics here